geoh5py/data/referenced_data.py

```diff
@@ -39,6 +43,42 @@ def __init__(self, **kwargs):
 
         super().__init__(**kwargs)
 
+    def copy(
+        self,
+        parent=None,
+        *,
+        clear_cache: bool = False,
+        mask: np.ndarray | None = None,
+        **kwargs,
+    ) -> Data:
+        """
+        Overwrite the copy method to ensure that the uid is set to None.
+        """
+        kwargs["uid"] = None
+
+        new_data = cast(
```

Comment:
I'm not against cast, but is there more "cleaned" way to ensure the type here?

Reply:
We would need to remove the typing on the base class, and type the sub-classes instead.
To be addressed on https://mirageoscience.atlassian.net/browse/GEOPY-2038